### PR TITLE
feat: tidy verbose telemetry service name

### PR DESCRIPTION
**Pull request by jamesstocktonj1**
## Feature or Problem
Tidy up the open telemetry attributes:
- Replace provider executable name in favour of the Provider Key for `service.name`
    - Provider Key is unique to each provider instance whereas the executable name is not
- Add Host ID to `host.id`
- Add Instance ID to `container.id`
- Move provider binary name to `container.image.name`

## Testing
Tested using some simple modifications to the http-server example which can be found in [this Gist](https://gist.github.com/jamesstocktonj1/b07495151c1142d122f66483fae3a878). The tracer collecter is Jaeger.

### Before
Screenshot from 2024-10-05 14-06-20

### After
Screenshot from 2024-10-05 14-06-44

### CODE REVIEW

File: observability.go [hunk 1]
```diff
@@ -124,16 +124,18 @@ func newLoggerProvider(ctx context.Context, config OtelConfig, serviceResource *
...
 		return nil, err
 	}
 
 	return resource.New(ctx,
 		resource.WithAttributes(
-			semconv.ServiceNameKey.String(filepath.Base(providerBinary)),
-			semconv.ServiceInstanceIDKey.String(name),
+			semconv.ServiceNameKey.String(hostData.ProviderKey),
+			semconv.HostIDKey.String(hostData.HostID),
+			semconv.ContainerIDKey.String(hostData.InstanceID),
```
Comment (joonas) [inline observability.go:137]: Let's use `ServiceInstanceIDKey` instead of `ContainerIDKey`

File: observability.go [hunk 2]
```diff
@@ -124,16 +124,18 @@ func newLoggerProvider(ctx context.Context, config OtelConfig, serviceResource *
...
+			semconv.ContainerImageNameKey.String(filepath.Base(providerBinary)),
```
Comment (joonas) [inline observability.go:138]: Let's use `ProcessExecutableNameKey` instead of `ContainerImageNameKey`

**Review by joonas — CHANGES_REQUESTED** [review]

**Comment by joonas** [comment]
Thanks for the PR, these are solid improvements, I just added a couple of comments in terms of adjusting the keys that we use for these to better align the semantics with the values.

@lxfontes may have further thoughts to add, I'll happily approve from my end once you have a chance the address the feedback.

File: observability.go (hunk 1 above)
Comment (jamesstocktonj1) [reply observability.go]: Done

**Review by joonas — APPROVED** [review]
Looks good on my part!

**Comment by joonas** [comment]
I'm leaving this open pending any feedback from @lxfontes.

Lucas, feel free to merge this if you don't have anything to add.

**Review by brooksmtownsend — APPROVED** [review]
LGTM to me as well, @lxfontes feel free to merge when approved!

**Pull request merged**